integration-tests/src/test/java/tachyon/master/MasterClientAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,201 @@
+    Assert.assertTrue(masterClient.getFileStatus(-1, "/file-nosasl") != null);
+    masterClient.disconnect();
+    Assert.assertFalse(masterClient.isConnected());
+    masterClient.connect();
+    Assert.assertTrue(masterClient.isConnected());
+    Assert.assertTrue(masterClient.getFileStatus(-1, "/file-nosasl") != null);
+    masterClient.close();
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+  }
+
+  @Test
+  public void simpleAuthenticationOpenCloseTest() throws Exception {
+    //simple authentication configure
+    setAuthenticationType(AuthType.SIMPLE);
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    mMasterInfo = mLocalTachyonCluster.getMasterInfo();
+    MasterClient masterClient =
+        new MasterClient(mMasterInfo.getMasterAddress(),
+            mExecutorService, mMasterTachyonConf);
+    Assert.assertFalse(masterClient.isConnected());
+    masterClient.connect();
+    Assert.assertTrue(masterClient.isConnected());
+    masterClient.user_createFile("/file-simple", "", Constants.DEFAULT_BLOCK_SIZE_BYTE, true);
+    Assert.assertTrue(masterClient.getFileStatus(-1, "/file-simple") != null);
+    masterClient.disconnect();
+    Assert.assertFalse(masterClient.isConnected());
+    masterClient.connect();
+    Assert.assertTrue(masterClient.isConnected());
+    Assert.assertTrue(masterClient.getFileStatus(-1, "/file-simple") != null);
+    masterClient.close();
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+  }
+
+  @Test
+  public void customAuthenticationOpenCloseTest() throws Exception {
+    //custom authentication configure
+    setAuthenticationType(AuthType.CUSTOM);
+    mMasterTachyonConf.set(Constants.TACHYON_AUTHENTICATION_PROVIDER_CUSTOM_CLASS,
+        NameMatchAuthenticationProvider.class.getName());
+
+    /**
+     * Using tachyon as loginUser for unit testing, tachyon* user is allowed to connect to
+     * Tachyon Master.
+     */
+    LoginUser.setLoginUser(LoginUser.createUserForTesting("tachyon"));
+
+    //start cluster
+    mLocalTachyonCluster.start();
+
+    mMasterInfo = mLocalTachyonCluster.getMasterInfo();
+    MasterClient masterClient =
+        new MasterClient(mMasterInfo.getMasterAddress(),
+            mExecutorService, mMasterTachyonConf);
+    Assert.assertFalse(masterClient.isConnected());
+    masterClient.connect();
+    Assert.assertTrue(masterClient.isConnected());
+    masterClient.user_createFile("/file-custom", "", Constants.DEFAULT_BLOCK_SIZE_BYTE, true);
+    Assert.assertTrue(masterClient.getFileStatus(-1, "/file-custom") != null);
+    masterClient.disconnect();
+    Assert.assertFalse(masterClient.isConnected());
+    masterClient.connect();
+    Assert.assertTrue(masterClient.isConnected());
+    Assert.assertTrue(masterClient.getFileStatus(-1, "/file-custom") != null);
+    masterClient.close();
+
+    //stop cluster
+    mLocalTachyonCluster.stop();
+  }
+
+  @Test
+  public void customAuthenticationDenyConnectTest() throws Exception {
```

Comment:
customAuthenticationDenyConnectTest has tested the connection failed case. It uses a NameMatchAuthenticationProvider authentication to verify only tahcyon user allowed to connect the Master service.